R/add.R

```diff
@@ -137,16 +137,22 @@ add_column <- function(.data, ..., .before = NULL, .after = NULL) {
 
   pos <- pos_from_before_after_names(.before, .after, colnames(.data))
 
+  end_pos <- ncol(.data) + seq_len(ncol(df))
+
   if (pos <= 0L) {
-    out <- cbind(df, .data)
+    indexes_orig <- seq_len(ncol(.data))
+    .data[end_pos] <- df
+    .data <- .data[c(end_pos, indexes_orig)]
   } else if (pos >= ncol(.data)) {
-    out <- cbind(.data, df)
+    .data[end_pos] <- df
   } else {
-    indexes <- seq_len(pos)
-    out <- cbind(.data[indexes], df, .data[-indexes])
+    indexes_before <- seq_len(pos)
+    indexes_after <- (length(indexes_before)+1):ncol(.data)
+    .data[end_pos] <- df
```

Comment:
Could you move this line and the line below after the `if` expression? Seems that we need to compute an `indexes` vector for all three cases, in the second case it's just a sequence. This makes all three cases more consistent and easier to troubleshoot.

Reply:
Moved the addition of the new column and the reordering outside of the `if`. I'm not sure if we can reduce the `if` to less than 3 branches. I thought about it but it seems like we need to check all of the cases to construct the correct index.